src/transformation/utils/symbols.ts

```diff
@@ -0,0 +1,70 @@
+import * as ts from "typescript";
+import * as lua from "../../LuaAST";
+import { getOrUpdate } from "../../utils";
+import { TransformationContext } from "../context";
+import { ReferencedBeforeDeclaration } from "./errors";
+import { markSymbolAsReferencedInCurrentScopes } from "./scope";
+import { getFirstDeclarationInFile } from "./typescript";
+
+const symbolIdCounters = new WeakMap<TransformationContext, number>();
+function nextSymbolId(context: TransformationContext): lua.SymbolId {
+    const symbolId = (symbolIdCounters.get(context) || 0) + 1;
+    symbolIdCounters.set(context, symbolId);
+    return symbolId as lua.SymbolId;
```

Comment:
Does the location of  SymbolId in lua make sense? Should we move it here?

Reply:
In that case we wouldn't be able to store it on `Identifier` node and have to use a WeakMap. That actually might be a good thing, but might require quite a lot of changes, so maybe leave it for a different PR?

Reply:
That's fine with me.